### DISCUSSION TITLE
Fooocus is using RAM instead GPU

### DISCUSSION THREAD

**Discussion by mchelequim** [lllyasviel/Fooocus#3302] [Q&A] [answered]
Hello, for some reason fooocus is using all my RAM freezing my pc some times, how do i config it to use my GPU? Im a begginer, please explain in details what should i do. :D im running Fooocus 2.5.0 (aditional information, while ram is at 98% GPU is at 1%)
hardware information : 16GB RAM, RTX 3060 12GB VRAM, Intel Core I59400F . 
I see similar questions but didn't find any answer that solve this problem.

**Comment by mashb1t** [comment, marked as answer]
You may edit your run.bat and add `--always-high-vram` at the end of the first line. Be advised that this is not necessary to fully load the GPU and fooocus already uses all necessary resources by default, this just more emphasizes on GPU usage.
Personally i'd recommend to use the default and ensure that Fooocus itself is located on an SSD.

**Reply by mchelequim** [reply]
I did, but it doesn’t work. I don’t know what to do, pc still freezing, even more when i use some refiner. It freezes right at the start when says: “loading models” I’m running from a ssd, but it’s not the same as the system, gonna try to move to drive C

**Reply by mashb1t** [reply]
@mchelequim you can also manually move the swap to a faster drive, optimally the same fast drive as Fooocus is on.
=> https://github.com/lllyasviel/Fooocus/blob/main/troubleshoot.md#system-swap (but manually set it to 30GB on the fast drive)

**Reply by mchelequim** [reply]
move to other drive dosen't work as i expected but i solve the problem with this video, is in portuguese but i gues it's undertandable, for anyone that's having the same problem, this could work: https://www.youtube.com/watch?v=aObA8m8oPGc

and thank you for the support :D

**Reply by WilliWantsToKnow** [reply]
I had the same problem. I had 16 GB of RAM and my PC was frozen the entire time. Generating an image on quality took me 40 - 50 min. I simply looked on eBay and bought a matching set of RAM sticks. Now I have 32 GB RAM. And now generating an image on quality only takes 6 min.
I know this probably isn´t the solution you where looking for but I highly recommend it. Just upgrade. It cost me 30 bucks on eBay. Try finding the exact same RAM sticks you already have in your PC. Having two different sticks can work, but I heard that it can cause problems or bottlenecks (RAM can only be as quick as the slowest one).

**Reply by mchelequim** [reply]
Thank you for the answer, for me is taking some seconds on quality, you should look for something else, it’s no lt normal wait all this time, even with Loras and refiners.

**Reply by WilliWantsToKnow** [reply]
What do you mean by seconds? 5, 10, 100 seconds? What time does it take you to generate a picture on quality, jpeg, and aspect ratio 1:1 on average?

**Reply by mchelequim** [reply]
50 to 60 seconds per image, on quality, standard styles, realistic stock photo Model aspct ratio 7:9

**Reply by mashb1t** [reply]
@mchelequim this is also the time which i'd expect a 3060 to take, so you can only optimize in model loading speeds, but not inference.

@WilliWantsToKnow even 6min is quite slow, but depends on your GPU.
